### PR TITLE
[ENH] - Update `create_dataframe_bins`

### PR DESCRIPTION
**Pull request by TomDonoghue**
This PR extends bin df creation to accept extra data columns. This can be useful when adding additional trial-level information to dataframes (for example, for splitter cell analyses in T3).

### CODE REVIEW

File: spiketools/tests/stats/test_anova.py [hunk 1]
```diff
@@ -26,27 +26,35 @@ def test_create_dataframe(tdata2d):
...
+    df = create_dataframe_bins(data2da, ['bin', 'fr'])
     assert isinstance(df, pd.DataFrame)
-    for ind, row in enumerate(data2d):
+    for ind, row in enumerate(data2da):
         df[df['trial'] == ind]['fr'].values == row
 
     # Create new data, where each bin value matches bin index
-    data2d = np.array([[0, 1, 2],
+    data2db = np.array([[0, 1, 2],
                        [0, 1, 2]])
```
Comment (maestapereira) [inline spiketools/tests/stats/test_anova.py:37]: Should we add a space here?

**Review by maestapereira — APPROVED** [review]
Looks great!

File: spiketools/stats/anova.py [hunk 2]
```diff
@@ -49,17 +50,21 @@ def create_dataframe(data, columns=None, drop_na=True, types=None):
...
+    bin_data : 2d or 3d array
         An array of data organized into pre-computed bins.
         If a 2d array, should be organized as [n_trials, n_bins].
         If a 3d array, should be organized as [n_trials, n_xbins, n_ybins].
-    columns : list of str
-        The column labels for the dataframe.
+    columns : list of str, optional
+        The column labels for the bin data.
+        Defaults to ['bin', 'fr'] for 1d bins or ['xbin', 'ybin' 'fr'] for 2d bins.
+    other_data : dict
```
Comment (claire98han) [inline spiketools/stats/anova.py:65]: Should `other_data` be optional too?

**Review by claire98han — APPROVED** [review]
Looks great to me too!

File: spiketools/tests/stats/test_anova.py (hunk 1 above)
Comment (TomDonoghue) [reply spiketools/tests/stats/test_anova.py:37]: yes! good catch!

File: spiketools/stats/anova.py (hunk 2 above)
Comment (TomDonoghue) [reply spiketools/stats/anova.py:65]: yes! I keep forgetting optional labels, thanks for finding!

**Pull request merged**